.circleci/config.yml

```diff
@@ -25,30 +14,43 @@ aliases:
     paths:
       - ~/.cache/yarn
 
+
+  - &restore_yarn_cache_node_14
+    keys:
+      - v1-yarn-cache-node_14-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - v1-yarn-cache-node_14-{{ .Branch }}
+      - v1-yarn-cache-node_14
+
+  - &save_yarn_cache_node_14
+    key: v1-yarn-cache-node_14-{{ .Branch }}-{{ checksum "yarn.lock" }}
+    paths:
+      - ~/.cache/yarn
+
   - &yarn_install
     name: Installing
     command: yarn --frozen-lockfile
 
 executors:
-  node_10:
-    docker:
-      - image: circleci/node:10.24.1
-    working_directory: *working_directory
   node_12:
     docker:
       # Keep it at `12.18.2` as otherwise there may be bugs with `graceful-fs` dependency
       # https://github.com/nodejs/node/issues/34529
+      # https://raw.githubusercontent.com/isaacs/node-graceful-fs/168bdb8f0bb3174e8499d4bc5878deead4172c39/polyfills.js
       - image: circleci/node:12.18.2
```

Comment:
I think we might want to decide how we align supported Node.js versions along the Node.js release cycle. It doesn't have to follow it but could be a "formula". As a suggestion:

1. We intend to support the "current" version(s) of Node.js 2-3 months after they are released
2. We support any "active" version(s) of Node.js
3. We stop supporting Node.js versions 3 months after they left their maintenance window

That way we can communicate a clear expectation also to customers which they can plan for. Which is important when they need to allocate time and money onto upgrades.

Would that make sense?

Reference: https://nodejs.org/en/about/releases/

Reply:
Sure, it does make sense, I will come up with a draft of all the points you highlighted above, present it to you and Nicola and we can see how we can incorporate it into our workflow specifically on this very nodejs project.

Reply:
Maybe with this @nkuehn could also be involved as it can be incorporated into the docs.

Reply:
Thanks for the heads up - please sync with @industrian once you have a "communicatable" policy description what your own major / minor versions mean and how they relate to nodejs support times.   

PS: The topic area was discussed at a much more generic level a while ago (I think the context was MC app kit versioning and compatibility), i could not find the discussion though now.  It would be very good if the policy for MC custom apps and the SDKs do not deviate.  See https://docs.commercetools.com/custom-applications/maintenance-guidelines   (which could profit from being easier to understand, too).